Support custom build_file_content

ahans opened the pull request:
This adds a string parameter `build_file_content` to the `deb_package` rule to allow for customization. One use case are packages that have files under `usr/include` or `usr/lib` that are not allowed in `hdrs` or `srcs`, for example `*.cmake`. With a custom `build_file_content`, we can fix that.

ahans committed:
Support custom build_file_content

This adds a string parameter `build_file_content` to the `deb_package`
rule to allow for customization. One use case are packages that have
files under `usr/include` or `usr/lib` that are not allowed in `hdrs` or
`srcs`, for example `*.cmake`. With a custom `build_file_content`, we
can fix that.

psigen commented:
This is a good idea!  We can also support the regular `buildfile` parameter.  Do you mind if I augment your PR to include that as well?

ahans commented:
> This is a good idea! We can also support the regular `buildfile` parameter.

I actually had the same thought, but was too lazy to implement that as well. ;-)

> Do you mind if I augment your PR to include that as well?

Sure, go ahead if you find the time!

ahans commented:
I ended up adding it myself. Let me know what you think!

The examples could also use some updating, it seems. At least one URL is apparently outdated.

ahans committed:
Add support for build_file

psigen committed:
Cleanups to error messages.

psigen committed:
Reorder arguments to support other exports.

I reordered the evaluation of the flags to put the buildfile and buildfile content first,
because I intended to have multiple `export_*` flags in the future for additional 
language targets and it's easier if they are in the other order.

Functionality should be identical to before.

psigen commented:
I did a quick refactor to support multiple export flags in the future, but it seemed to be working for me!

> The examples could also use some updating, it seems. At least one URL is apparently outdated.

It's a bit worse than that, the debian package repository layout loves to randomly change, and the by-hash paths
I used at the time of initial implementation appear to have shifted somewhat.  I should make a separate PR
to correct for this.
